.gulp/gulpfile.iced

```diff
@@ -21,14 +23,34 @@ task 'init', "" ,(done)->
   done()
 
 # Run language-specific tests:
-task 'test', 'type check generated code', [], (done) ->
+task 'test', '', ['test/typecheck', 'test/nodejs-unit', 'test/chrome-unit'], (done) ->
+  done();
+
+task 'test/typecheck', 'type check generated code', [], (done) ->
   await execute "tsc -p ./test/tsconfig.generated.json", defer _
   done();
 
-task 'test', 'run unit tests', [], (done) ->
+task 'test/nodejs-unit', 'run nodejs unit tests', [], (done) ->
   await execute "mocha", defer _
   done();
 
+task 'test/chrome-unit', 'run browser unit tests', [], (done) ->
+  count = 2;
+  testServer = child_process.spawn("node", ["./startup/www.js"], { cwd: "./node_modules/@microsoft.azure/autorest.testserver" })
+  webpackDevServer = child_process.spawn("./node_modules/.bin/webpack-dev-server", [], {})
+  onData = (data) ->
+    count--
+    if count == 0
+      runMochaChrome(testServer, webpackDevServer, done)
+
+  runMochaChrome = () ->
+    await execute "./node_modules/.bin/mocha-chrome http://localhost:8080", defer _;
+    testServer.kill();
+    webpackDevServer.kill();
+
+  testServer.stdout.on('data', onData)
+  webpackDevServer.stdout.on('data', onData)

```

Comment:
Since you're passing onData to both of these, does that mean that you're just waiting for both of them to finish, and then you're going to runMochaChrome after that?

Reply:
When the server writes to stdout, we assume it's ready to receive HTTP requests.